src/ansys/templates/python/solution/{{cookiecutter.__project_name_slug}}/.github/workflows/build-release.yml

```diff
@@ -65,19 +67,137 @@ jobs:
           retention-days: "7"
           modern-installation: "true"
 
+  run-tests:
+    name: Run tests
+    runs-on: [windows-latest]
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
+      fail-fast: false
+    steps:
+      - name: Run tests
+        uses: ansys-internal/solution-applications-actions/run-tests@v11
+        with:
+          python-version: {{ "${{ matrix.python-version }}" }}
+          gh-token: {{ "${{ secrets.WORKFLOW_TOKEN }}" }}
+          check-coverage-ratio: false
+
+  check-licenses:
+    if: |
+      github.event_name == 'push' && github.ref == 'refs/heads/main'
+    name: Check Licenses
+    runs-on: [ubuntu-latest]
+    timeout-minutes: 10
+    steps:
+      - name: Check Licneses
+        uses: ansys-internal/solution-applications-actions/check-licenses@v11
+        with:
+          gh-token: {{ "${{ secrets.WORKFLOW_TOKEN }}" }}
+          python-version: {{ "${{ env.MAIN_PYTHON_VERSION }}" }}
+
+  package-solution:
+    name: Package Solution
+    needs: build
+    runs-on: windows-latest
+    steps:
+      - name: Package Solution
+        uses: ansys-internal/solution-applications-actions/package-solution@v11
+        with:
+          desktop-utility-version: "v1"
+          gh-token: {{ "${{ secrets.WORKFLOW_TOKEN }}" }}
+          python-version: {{ "${{ env.MAIN_PYTHON_VERSION }}" }}
+          solution-display-name: "{{ cookiecutter.solution_display_name }}"
+          solution-name: "{{ cookiecutter.__project_name_slug }}"
+          solutions-pypi-token: {{ "${{ secrets.SOLUTIONS_PRIVATE_PYPI_PAT }}" }}
+
+  generate-sbom:
+    name: Generate SBOM
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Generate SBOM
+        uses: ansys-internal/solution-applications-actions/generate-sbom@v11
+        with:
+          lockfile-path: "./poetry.lock"
+          sbom-name: {{ "${{ env.SBOM_NAME }}" }}
+          gh-token: {{ "${{ secrets.WORKFLOW_TOKEN }}" }}
+          python-version: {{ "${{ env.MAIN_PYTHON_VERSION }}" }}
+
+  validate-sbom:
+    if: |
+      github.event_name == 'push'
+    name: Validate SBOM
+    runs-on: [ubuntu-latest]
+    timeout-minutes: 10
+    needs: generate-sbom
+    steps:
+      - name: Validate SBOM
+        uses: ansys-internal/solution-applications-actions/validate-sbom@v11
+        with:
+          sbom-artifact: {{ "${{ env.SBOM_NAME }}" }}
+          gh-token: {{ "${{ secrets.WORKFLOW_TOKEN }}" }}
+
   release:
     if: |
       github.event_name == 'release'
     name: Release
-    needs: [check-copyright, check-code-style, build]
+    needs: [build-doc, check-copyright, check-code-style, build, run-tests, check-licenses, package-solution, generate-sbom, validate-sbom]
     runs-on: [ubuntu-latest]
     timeout-minutes: 10
     steps:
       - name: Release
-        uses: Solution-Applications/actions/release@v5
+        uses: ansys-internal/solution-applications-actions/release-to-private-pypi@v11
```

Comment:
I think that we should not be doing this by default - generally, the solution wheel isn't used for anything.

Reply:
What do you suggest? Should I remove the release action?

Reply:
I suggest either removing it or hiding it behind an optional input that defaults to false. For solutions, we already upload the built package as part of the release [here](https://github.com/ansys/ansys-templates/blob/328d4ea8c6496b173bfcefa4bb88dec9447e71a7/src/ansys/templates/python/solution/%7B%7Bcookiecutter.__project_name_slug%7D%7D/.github/workflows/release-please.yml#L49-L58)

In situations where customer access is needed, the built package is the part that is useful to upload to Artifactory. We did this for NVH [here](https://github.com/ansys-internal/em-nvh-mapper/blob/9e94e173a7c7fd1a5d2fb678af7a723feb529372/.github/workflows/build-release.yml#L162-L177)